### PR TITLE
Add migrate-tags.py script for migrating Smarty tags

### PR DESCRIPTION
**Pull request by briansull**
Also added a test case for the script

@jashook @ramarag @kyulee1 PTAL

### CODE REVIEW

**Comment by briansull** [comment]
Output from do-test.cmd:
D:\fxkit\coreclr\tests\scripts\test>..\migrate-tags.py Tests.lst TestsOld.lst
Starting migrate-tags: Last Updated - 10-Mar-16
- - - - - - - - - - - - - - - - - - - - - - - - - - - -
Parsing: Tests.lst
8475 tests found in Tests.lst

Parsing: TestsOld.lst
5877 tests found in TestsOld.lst

Migrating the tags
2859 new tests found and tagged as NEW

Writing: Tests.lst
Wrote Tests.lst with 8475 tests

**Comment by jashook** [comment]
Skip ci test would have been useful here. The tests shouldn't need to be run.

File: tests/scripts/migrate-tags.py [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+    test_number = None
+    test_fullpath = None
+
+    # This could also be a dictionary, but we want to preserve the order
+    # that we encounter the entries and we only have a small number of keys
+    # and python supports casting a list into a dictionary wehn you need to
+    # Also consider using class OrderDict
+    test_properties = []
+
+    with open(listfile, 'r') as file:
```
Comment (jashook) [inline tests/scripts/migrate-tags.py]: 'r' is not useful here

Comment (jashook) [inline tests/scripts/migrate-tags.py]: 'r' is not needed here

File: tests/scripts/migrate-tags.py [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+                        # so we record it's value here.  
+                        if test_fullpath is not None:
+                            fatal('logic error - fullpath is already set');
+                        test_fullpath = value
+                    
+                    tup = (key, value)
+                    test_properties.append(tup)                    
+                else:
+                    # we didn't have a 'key=value' line
+                    # will will switch to expecting_testname
```
Comment (jashook) [inline tests/scripts/migrate-tags.py]: typo: will duplicated

File: tests/scripts/migrate-tags.py [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+                # have to get the testnumber from the end using [-1]
+                if len(split_line) == 1:
+                    fatal('syntax error - missing _' + line);
+    
+                # elements in split_line are numbered [ 0, 1, ... -2, -1 ]
+                test_number_str = split_line[-1]
+                if len(split_line) == 2: 
+                    test_name = split_line[0] 
+                else:
+                    # test_name = '_'.join(split_line[0:len(split_line)-1])
```
Comment (jashook) [inline tests/scripts/migrate-tags.py]: Was this line supposed to be removed?

File: tests/scripts/migrate-tags.py [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+                if len(split_line) == 1:
+                    fatal('syntax error - missing _' + line);
+    
+                # elements in split_line are numbered [ 0, 1, ... -2, -1 ]
+                test_number_str = split_line[-1]
+                if len(split_line) == 2: 
+                    test_name = split_line[0] 
+                else:
+                    # test_name = '_'.join(split_line[0:len(split_line)-1]) 
+                    test_name = '_'.join(split_line[0:-1])
```
Comment (jashook) [inline tests/scripts/migrate-tags.py]: Thank you for making this change, it is more readable because it follows convention.

**Comment by jashook** [comment]
LGTM with a few nits

File: tests/scripts/migrate-tags.py (hunk 3 above)
Comment (briansull) [inline tests/scripts/migrate-tags.py]: Yes after I retested that it worked, which I did
WIll remove

File: tests/scripts/migrate-tags.py (hunk 2 above)
Comment (briansull) [inline tests/scripts/migrate-tags.py]: fixed

File: tests/scripts/migrate-tags.py [hunk 5]
```diff
@@ -0,0 +1,327 @@
+﻿#!/usr/bin/env python
```
Comment (kyulee1) [inline tests/scripts/migrate-tags.py]: Do we need a license banner at the top?

File: tests/scripts/migrate-tags.py (hunk 1 above)
Comment (briansull) [inline tests/scripts/migrate-tags.py]: ok I will remove it

File: tests/scripts/migrate-tags.py (hunk 5 above)
Comment (briansull) [inline tests/scripts/migrate-tags.py]: I will add one

**Pull request merged**